### PR TITLE
Udes feature/514

### PR DESCRIPTION
**Pull request by CallumCoombes**

### CODE REVIEW

File: models/trailer_info.py [hunk 1]
```diff
@@ -40,3 +40,22 @@ def create(self, values):
...
+        # Create a dict of all the fields
+        data = {'trailer_num': self.trailer_num, 'trailer_ident': self.trailer_ident,
+                'trailer_license': self.trailer_license, 'trailer_driver': self.trailer_driver}
+
+        return data
+
+    def get_info(self, **kwargs):
+        """ Return a list with the information of each picking in self.
+        """
+
```
Comment (ws-mps) [inline models/trailer_info.py]: Remove empty line.

File: models/stock_picking.py [hunk 2]
```diff
@@ -2,73 +2,80 @@
...
+        if self.u_requires_transport and (not fields_to_fetch or 'u_transport_id' in fields_to_fetch):
+            if self.u_transport_id:
+                # Get transport info and add it to data
+                data['u_transport_id'] = {'u_vehicle_sequence': self.u_vehicle_sequence,
+                                          'u_vehicle_description': self.u_vehicle_description,
+                                          'u_license_plate': self.u_license_plate,
+                                          'u_driver_name': self.u_driver_name}
+            else:
+                # for the case with no information, the front end expects an empty list
+                data['u_transport_id'] = []
```
Comment (ws-mps) [inline models/stock_picking.py]: empty list or empty dictionary?

File: models/transport.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    _sql_constraints = [
+        ('picking_id_uniq', 'unique(picking_id)', 'Only one transport information is allowed per picking.'),
+    ]
+
+    def _update_picking_data(self):
+        """ Update u_transport_id of the related picking if
+            it is not set at the picking.
+        """
+        self.ensure_one()
+        from_picking = self.env.context.get('from_picking', False)
```
Comment (ws-mps) [inline models/transport.py]: This is not used, not needed.

Comment (CallumCoombes) [reply models/transport.py]: Line removed

File: models/stock_picking.py (hunk 2 above)
Comment (CallumCoombes) [reply models/stock_picking.py]: After discussion, code was changed to only return fields that have been filled in

**Review by ws-mps — APPROVED** [review]

**Review by readikus — APPROVED** [review]

**Pull request merged**